File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/AppModule.java
package at.tugraz.tc.cyfile;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import at.tugraz.tc.cyfile.injection.ApplicationContext;
import at.tugraz.tc.cyfile.logging.CyFileLogger;
import dagger.Module;
import dagger.Provides;

/**
 * Dagger application module
 * Created by cobri on 3/21/2018.
 */

@Module
public class AppModule {

    private Application mApplication;

    public CyFileLogger logger;

    public AppModule(Application mApplication, CyFileLogger logger) {
        this.mApplication = mApplication;
        this.logger = logger;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return mApplication;
    }

    @Provides
    @ApplicationContext
    Context provideContext() {
        return mApplication;
    }

    @Provides
    @Singleton
    CyFileLogger provideLogger() { return logger; }

}

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/AppModule.kt
package at.tugraz.tc.cyfile

import android.app.Application
import android.content.Context

import javax.inject.Singleton

import at.tugraz.tc.cyfile.injection.ApplicationContext
import at.tugraz.tc.cyfile.logging.CyFileLogger
import dagger.Module
import dagger.Provides

/**
 * Dagger application module
 * Created by cobri on 3/21/2018.
 */

@Module
class AppModule(private val mApplication: Application, var logger: CyFileLogger) {

    @Provides
    @Singleton
    internal fun provideApplication(): Application {
        return mApplication
    }

    @Provides
    @ApplicationContext
    internal fun provideContext(): Context {
        return mApplication
    }

    @Provides
    @Singleton
    internal fun provideLogger(): CyFileLogger {
        return logger
    }

}
